### ISSUE TITLE
Refactor codebase to C++

### ISSUE THREAD

**Issue by madcowswe** [odriverobotics/ODrive#29]
Progressively move the code over to C++. At the same time compartmentalise and organise, make more modular.

**Comment by madcowswe** [comment]
```C++
//Outside axis:
    //command handler
    //callback dispatch

class Axis {
public:
    //thread/os/system management
        //timing log
        //thread id
        //etc.
    //state machine
        //control mode
        //control_en/calib_ok
        //error state
    //motor
        //current controller
            //contains rotor phase logic
        //motor level calibration routines
        //low_level (implementation specifics)
            //DRV driver
            //adc callback handling
            //pwm queueing
    //rotor estimator
        //kick out rotor phase logic
    //pos/vel controller
    //step/dir handler
}
```

**Comment by samuelsadok** [comment]
addressed in #136
